docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+
+![Sequence diagram for parsing list command](./diagrams/ListParseSequenceDiagram.jpg)
+
+#### Calendar feature
+(WIP)
+
+#### Repeat feature
+
+The repeat feature on the program allows for the user to be able to make certain events repeat several times over a defined time period.
+For instance, the user can request for a personal event to be repeated monthly for four months. To run the program, the user will need to key in the command `repeat [event type] [index] [timeUnit] [count]`
+
+|Argument| Description |
+|--------|----------|
+|event type|What type of event is to be repeated? Accepted arguments are `personal`, `timetable` or `zoom`|
+|index|Index number of the event to be repeated that is stored on the Event List|
+|timeUnit|For what unit of time each event is to be repeated. Accepted arguments are `daily`, `weekly` and `monthly`|
+|count|Integer indicating how many times the event is to be repeated|
+
+For example, to repeat a personal event located at index 2 for 3 days, the user will key in this: `repeat personal 2 daily 3`.
+
+The following sequence diagram shows the overall process of repeating an event:
+
+![Sequence Diagram for Repeat Command](./diagrams/RepeatScenario.jpg)
+
+The steps to repeating the command will be described in the following sections. 
+
+##### Step 1: Obtaining the event information
+
+The repeat command first locates the event that is to be repeated along with its starting date.
+Next, it creates an empty ArrayList called `repeatEventList` that will contain all repeated events.
+
+![Sequence Diagram for Repeat Command step 1](./diagrams/repeatstep1.jpg)
+
+##### Step 2: Incrementing the Original Date
+
+The repeatCommand object will now call upon Event to increment the date by a fixed unit of time. In the following diagram, it shows how the program flow will work should a monthly repeat increment be set. 
+The increment date will be assigned the name `repeatDate`
+
+![Sequence Diagram for Repeat Command step 2](./diagrams/repeatstep2.jpg)
+
+##### Step 3: Set Repeat type
+
+The repeatCommand now will set the original Event to have a repeat type setting. In the diagram's example, it will be set to `monthly`
+
+![Sequence Diagram for Repeat Command step 3](./diagrams/repeatstep3.jpg)
+
+##### Step 4: Clone the event
+
+The original event is now cloned as shown in the following sequence diagram.
+
+![Sequence Diagram for Repeat Command step 4](./diagrams/repeatstep4.jpg)
+
+##### Step 5: Add the cloned activity
+
+The cloned activity will now have its date set to be `repeatDate`. This new activity is now added into the `repeatEventList` as shwon in the following diagram.
+
+![Sequence Diagram for Repeat Command step 5](./diagrams/repeatstep5.jpg)
+
+##### Step 6: Repeat steps 2 to 5
+
+Steps 2 to 5 are repeated up till the number specified by `count` as shown in the loop.
+
+![Sequence Diagram for Repeat Command step 6](./diagrams/repeatstep6.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/39303087/97660673-3d987c80-1aad-11eb-9139-e862d4202c2f.png)

The arrows (at the end) are supposed to be connected I think? May want to consider joining it :)